packages/runtime/plugin-runtime/src/rsc/server.ts

```diff
@@ -1 +1,37 @@
 export * from '@modern-js/render/rsc';
+
+export async function processRSCStream(
+  rscStream: ReadableStream,
+  controller: ReadableStreamDefaultController<Uint8Array>,
+  encoder: TextEncoder,
+) {
+  try {
+    const reader = rscStream.getReader();
+    const decoder = new TextDecoder('utf-8', { fatal: true });
+
+    while (true) {
+      const { done, value } = await reader.read();
+      if (done) break;
+
+      try {
+        const chunk = JSON.stringify(decoder.decode(value, { stream: true }));
+        const scriptTag = `<script>(self.__FLIGHT_DATA||=[]).push(${chunk})</script>`;
```

Comment:
And I wonder if `__FLIGHT_DATA` is consumed by React directly ?

Reply:
It will not be consumed directly by React, we use `rsc-html-stream` to consume `__FLIGHT_DATA`, and get the payload stream.